.github/workflows/CI_pull_request.yml

```diff
@@ -6,6 +6,14 @@ on:
   schedule:
       - cron: '15 2 * * TUE'  # Also run the full test suite for PRs on the main branch every Tuesday night
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - uses: pre-commit/action@v3.0.1
```

Comment:
> However, the pre-commit/action is marked in ["maintenance-only mode"](https://github.com/pre-commit/action) which makes me hesitant to introduce it.

It's in that mode because the maintainer wants people to use pre-commit.ci, which I don't like for multiple reasons. However, the action has been proven to Just Work even if it's in "maintenance-only mode", and has done so for a long time. And as you say, if it sometime stops working, it's easy to replace. (In fact, I have a [toy project](https://github.com/akx/preco) to reimplement pre-commit in Rust, anyway...)

> We can leave out the pre-commit integration altogether

The pre-commit configuration in this PR also sets up some of pre-commit's internal hooks for maintaining more hygiene (line endings, etc), and one could set up [a whole bunch of other things there too](https://github.com/home-assistant/core/blob/dev/.pre-commit-config.yaml). (In fact, #60 was borne out of having `crate-ci/typos` in the pre-commit configuration for a while.)

Ruff's editor integration is of course fine and dandy for running Ruff. Running these lints in CI via pre-commit has proven to be a good solution for many projects I've contributed to and/or maintained.

Reply:
Having now fully understood your setup here, I agree that this is the best approach. Let's keep the pre-commit/action.